**Simpson's Paradox is a statistical phenomenon where a trend that appears in different groups of data disappears or reverses when the groups are combined. This paradox often occurs because of lurking or confounding variables that skew the results when aggregated.**

In [1]:
import pandas as pd

In [2]:
# Step 1: Create the dataset
data = {
'Symptom Severity': ['Mild', 'Mild', 'Severe', 'Severe'],
'Medication': ['A', 'B', 'A', 'B'],
'Total Patients': [90, 10, 10, 90],
'Recovered Patients': [81, 9, 2, 63]
}

In [3]:
df = pd.DataFrame(data)

In [4]:
df

,Symptom Severity,Medication,Total Patients,Recovered Patients
0,Mild,A,90,81
1,Mild,B,10,9
2,Severe,A,10,2
3,Severe,B,90,63


In [5]:
# Step 2: Calculate recovery rates
df['Recovery Rate'] = df['Recovered Patients'] / df['Total Patients'] * 100
print("Recovery rates within each severity group:\n", df)

Recovery rates within each severity group:
   Symptom Severity Medication  Total Patients  Recovered Patients  \
0             Mild          A              90                  81   
1             Mild          B              10                   9   
2           Severe          A              10                   2   
3           Severe          B              90                  63   

   Recovery Rate  
0           90.0  
1           90.0  
2           20.0  
3           70.0  


In [8]:
print(total_recovered_a)
print(total_patients_a)

83
100


In [7]:
# Step 3: Calculate overall recovery rates
total_patients_a = df[df['Medication'] == 'A']['Total Patients'].sum()
total_recovered_a = df[df['Medication'] == 'A']['Recovered Patients'].sum()
overall_recovery_rate_a = total_recovered_a / total_patients_a * 100

total_patients_b = df[df['Medication'] == 'B']['Total Patients'].sum()
total_recovered_b = df[df['Medication'] == 'B']['Recovered Patients'].sum()
overall_recovery_rate_b = total_recovered_b / total_patients_b * 100

print(f"\nOverall recovery rate for Medication A: {overall_recovery_rate_a:.2f}%")
print(f"Overall recovery rate for Medication B: {overall_recovery_rate_b:.2f}%")



Overall recovery rate for Medication A: 83.00%
Overall recovery rate for Medication B: 72.00%


In [9]:
# Step 4: Interpret the results
if overall_recovery_rate_a > overall_recovery_rate_b:
    print("\nMedication A appears more effective overall.")
else:
    print("\nMedication B appears more effective overall.")


Medication A appears more effective overall.


In [10]:
df['Symptom Severity'].unique()

array(['Mild', 'Severe'], dtype=object)

In [11]:
for severity in df['Symptom Severity'].unique():
    severity_data = df[df['Symptom Severity'] == severity]
    print(f"Severity: {severity}")
    print(severity_data)


Severity: Mild
  Symptom Severity Medication  Total Patients  Recovered Patients  \
0             Mild          A              90                  81   
1             Mild          B              10                   9   

   Recovery Rate  
0           90.0  
1           90.0  
Severity: Severe
  Symptom Severity Medication  Total Patients  Recovered Patients  \
2           Severe          A              10                   2   
3           Severe          B              90                  63   

   Recovery Rate  
2           20.0  
3           70.0  


In [12]:
# Step 5: Compare the individual severity groups
print("\nWhen broken down by symptom severity:")
for severity in df['Symptom Severity'].unique():
    severity_data = df[df['Symptom Severity'] == severity]
    better_medication = severity_data.loc[severity_data['Recovery Rate'].idxmax(), 'Medication']
    print(f"- In {severity} cases, Medication {better_medication} has a higher recovery rate.")


When broken down by symptom severity:
- In Mild cases, Medication A has a higher recovery rate.
- In Severe cases, Medication B has a higher recovery rate.


**Explanation:**

Step 1: We create a DataFrame with the given data.

Step 2: We calculate the recovery rate for each medication within each symptom severity group.

Step 3: We then calculate the overall recovery rate for each medication across all groups.

Step 4: We determine which medication seems more effective overall.

Step 5: Finally, we compare the recovery rates within each symptom severity group to see if one medication consistently outperforms the other.



Recovery Rates within Each Severity Group:

Mild Symptoms:

Medication A: 90%

Medication B: 90%

Severe Symptoms:

Medication A: 20%

Medication B: 70%

Overall Recovery Rates (ignoring severity):

Medication A: 83%

Medication B: 72%

Interpretation:

Within the severe group, Medication B is clearly better.

Within the mild group, recovery rates are equal.

Overall, ignoring severity, Medication A appears better.

This is a classic Simpson’s Paradox: the overall trend (A looks better) reverses or hides the group-level trend (B is better in severe cases), showing why it’s important to consider subgroups in analysis.